pkg/secret/certificates.go

```diff
@@ -65,7 +65,7 @@ var (
 type Certificates []*Certificate
 
 // NewCertificatesForInitialControlPlane returns a list of certificates configured for a control plane node.
-func NewCertificatesForInitialControlPlane() Certificates {
+func NewCertificatesForInitialControlPlane(config *bootstrapv1.KThreesConfigSpec) Certificates {
```

Comment:
is there something we can do for "join control planes" to allow existing clusters to use this feature?

Reply:
The existing clusters etcd cert is stored in target cluster vm. There is no good way to export that cert back to mgmt cluster, unless user ssh vm, export file and create secret manually. Besides, k3s ca cert https://docs.k3s.io/cli/certificate is a bit different than what we create in k3s cp provider, adding possibility of human error and k3s versioning, which makes things more complex...

I would prefer adding flags in kcp cr, let it tell if it enabled etcd connection. New created cluster can enable it by set flag.

Reply:
renaming KThreesConfigSpec helper method to `IsEtcdEmbedded`. Differentiate with controlplane method `IsEtcdManaged`

IsEtcdManaged shall be IsEtcdEmbedded && etcdCASecret exists.